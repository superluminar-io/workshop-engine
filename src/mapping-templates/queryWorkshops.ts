export const request = `
  #if( $ctx.identity.resolverContext.role == 'ADMIN' )
    {
      "version": "2017-02-28",
      "operation": "Query",
      "query": {
        "expression": "node = :node",
        "expressionValues": {
          ":node": {
            "S" : "workshop"
          }
        },
      },
      "index": "byNode",
      "limit": $util.defaultIfNull($ctx.args.limit, 20),
      "nextToken": $util.toJson($util.defaultIfNullOrBlank($ctx.args.nextToken, null))
    }
  #else
    {
      "version": "2017-02-28",
      "operation": "Query",
      "query": {
        "expression": "node = :node",
        "expressionValues": {
          ":node": {
            "S" : "workshop"
          }
        },
      },
      "filter": {
        "expression": "contains(#attendees, :attendee)",
        "expressionNames": {
          "#attendees": "attendees",
        },
        "expressionValues": {
          ":attendee": {
            "S": "$ctx.identity.resolverContext.emailAddress"
          }
        }
      },
      "index": "byNode",
      "limit": $util.defaultIfNull($ctx.args.limit, 20),
      "nextToken": $util.toJson($util.defaultIfNullOrBlank($ctx.args.nextToken, null))
    }
  #end
`;

export const response = `
  {
    "items": $util.toJson($ctx.result.items),
    "nextToken": $util.toJson($util.defaultIfNullOrBlank($context.result.nextToken, null))
  }
`;
