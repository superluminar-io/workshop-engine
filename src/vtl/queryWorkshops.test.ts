import Parser from 'appsync-template-tester';
import { request, response } from './queryWorkshops';

describe('request mapping', () => {
  describe('user has admin role', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const context = {
        identity: {
          resolverContext: {
            emailAddress: 'me@example.com',
            role: 'ADMIN',
          },
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('user has not admin role', () => {
    it('should match snapshot', () => {
      const parser = new Parser(request);

      const context = {
        identity: {
          resolverContext: {
            emailAddress: 'me@example.com',
            role: 'ATTENDEE',
          },
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});

describe('response mapping', () => {
  describe('result w/ next token', () => {
    it('should match snapshot', () => {
      const parser = new Parser(response);

      const context = {
        result: {
          items: [
            {
              any: 'data',
            },
            {
              any: 'data',
            },
          ],
          nextToken: 'abc',
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('result w/o next token', () => {
    it('should match snapshot', () => {
      const parser = new Parser(response);

      const context = {
        result: {
          items: [
            {
              any: 'data',
            },
            {
              any: 'data',
            },
          ],
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });
});
