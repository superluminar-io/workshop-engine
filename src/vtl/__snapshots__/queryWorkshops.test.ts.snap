// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping user has admin role should match snapshot 1`] = `
Object {
  "index": "byNode",
  "limit": 20,
  "nextToken": null,
  "operation": "Query",
  "query": Object {
    "expression": "node = :node",
    "expressionValues": Object {
      ":node": Object {
        "S": "workshop",
      },
    },
  },
  "scanIndexForward": false,
  "version": "2017-02-28",
}
`;

exports[`request mapping user has not admin role should match snapshot 1`] = `
Object {
  "filter": Object {
    "expression": "contains(#attendees, :attendee)",
    "expressionNames": Object {
      "#attendees": "attendees",
    },
    "expressionValues": Object {
      ":attendee": Object {
        "S": "me@example.com",
      },
    },
  },
  "index": "byNode",
  "limit": 20,
  "nextToken": null,
  "operation": "Query",
  "query": Object {
    "expression": "node = :node",
    "expressionValues": Object {
      ":node": Object {
        "S": "workshop",
      },
    },
  },
  "scanIndexForward": false,
  "version": "2017-02-28",
}
`;

exports[`response mapping result w/ next token should match snapshot 1`] = `
Object {
  "items": Array [
    Object {
      "any": "data",
    },
    Object {
      "any": "data",
    },
  ],
  "nextToken": "abc",
}
`;

exports[`response mapping result w/o next token should match snapshot 1`] = `
Object {
  "items": Array [
    Object {
      "any": "data",
    },
    Object {
      "any": "data",
    },
  ],
  "nextToken": null,
}
`;
